### ISSUE TITLE
The items lines and Authoriser Quantity is 0 in remote authorisation

### ISSUE THREAD

**Issue by santoshkhanal12** [msupply-foundation/open-msupply#3528]
## What went wrong? 😲
 
The items lines and Authoriser Quantity is 0 in remote authorisation

![Screenshot 2024-04-03 at 15 55 35](https://github.com/msupply-foundation/open-msupply/assets/117138852/bcc95e48-a502-4509-9f48-d7b20c0cdac6)

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Set up Remote authorisation between two OMS site
2. Create Internal Order, add items by masterlist
3. Enter the values and send the IO
4. Sync
5. Now check Remote authorisation and you will see that Line number and authoriser quantity is set to 0 (BUG)

## Your environment 🌱

- Open mSupply Version: v1.7-00-RC4
- Legacy mSupply Central Server Version:v7-14-00-dev3

- Platform:

- Database type:

**Comment by Chris-Petty** [comment]
Can't replicate in dev 😱  
![image](https://github.com/msupply-foundation/open-msupply/assets/7684221/4dc61c4c-ca8e-467b-b20d-18ddc8d91cbf)

**Comment by nisha-dangol** [comment]
Checked in v1.7-rc5; as long as the request if from oms-store, I can replicate the issue: the authoriser quan is 0 by default:
<img width="229" alt="image" src="https://github.com/msupply-foundation/open-msupply/assets/59193490/0c2a2849-ceb1-443b-a450-7796e7bbe19f">

**Comment by Chris-Petty** [comment]
I had a peek at the code because I was pretty sure that'd simply confirm the case @nisha-dangol and yea:

https://github.com/msupply-foundation/open-msupply/blob/301e6628b7a29f439ca8c83c9f263a2396c2f44c/server/service/src/processors/transfer/requisition/create_response_requisition.rs#L218

[Legacy:](https://github.com/msupply-foundation/msupply/blob/a7b059ac30af34566848fd922a15ce75b649d3d7/Project/Sources/Methods/requisitionLineCreateResponse.4dm#L58-L60)

```4dm	
If (pref_get_boolean("store_preferences"; "shouldAuthoriseResponseRequisition"; [requisition]store_ID))  // Store preference : Use remote authorisation for response requisition turned on
  [requisition_line]approved_quantity:=[requisition_line]Cust_stock_order
End if 
```

So because the response requisition is created by OMS rather than the central server, it handles it a little bit differently. @andreievg @mark-prins @richardmoizeau I guess the question is simply what would our existing OMS authorisation module users want? 

Perhaps OMS can always just assign the ordered value to the approved quantity presuming it's only relevant if authorisation is needed.

**Comment by andreievg** [comment]
I haven't quite read through the issue, but saw severity low and a hot fix, is that right ?

**Comment by Chris-Petty** [comment]
@andreievg For release we just been trying to fix as many bugs as we can to raise the bar. This one should be trivial to fix but needs a decision on UX. In a nutshell if central server generates the transfer records the authoriser quantity will automatically be the requested quantity, if OMS generates it the authoriser quantity is 0. There is a button in the web app to "SET AUTHORISER QUANTITY TO REQUESTED" so the work around is pretty straight forward, hence low priority. 

I think a PO (@mark-prins ?) should decide which behaviour is correct - arguably 0 as a default has advantage if the authoriser just wants to reject everything, though there is a `DENY` button for that.

**Comment by lache-melvin** [comment]
Bumping - has/can a design decision be made here?

**Comment by andreievg** [comment]
Unassigning for now, waiting on design deicision

**Comment by roxy-dao** [comment]
Should keep at 0 for OG and OMS since there is a button to set authoriser quantity to requested

**Comment by Chris-Petty** [comment]
In that case this issue can be closed/moved to mSupply right?

**Comment by roxy-dao** [comment]
To auth repo >.>? Was going move it towards the end of the week

**Comment by Chris-Petty** [comment]
Nah to mSupply - the Authorised Quantity default value is populated when the transfer records are generated. When Legacy Desktop is the receiving party of the order, Legacy Central Server generates the requisition lines approved quantity with the requested quantity rather than 0. Auth repo just renders the record data.

**Comment by roxy-dao** [comment]
> Nah to mSupply - the Authorised Quantity default value is populated when the transfer records are generated. When Legacy Desktop is the receiving party of the order, Legacy Central Server generates the requisition lines approved quantity with the requested quantity rather than 0. Auth repo just renders the record data.

Oh then please do!
